site/_posts/2021-09-16-how-to-authenticate-your-react-app.md

```diff
@@ -0,0 +1,1087 @@
+There are two important things to call out here:
+
+- The `redirect_uri` which is the express server route that you configured as an authorized redirect URI initially. This is present because we track it throughout the process. If it ever differs from what we expect, it's an error condition.
+- `form`, which is the set of parameters FusionAuth expects.
+
+The OAuth token endpoint requires form encoded data, which is why we are using the `form` option in our request call.
+
+You can check the FusionAuth docs to see what every endpoint expects to be sent, but here’s the gist of this request:
+
+- The `client_id` and `client_secret` identify and authenticate our app, like a username and password. They are drawn from the FusionAuth configuration.
+- The `code` is the Authorization Code we got from FusionAuth.
+- The `grant_type` tells FusionAuth we’re using the The OAuth Authorization Code Grant (as opposed to an Implicit Grant or any other grant).
+- The `redirect_uri`  is a value from the "Authorized Redirect URLs" configuration. Requiring this is another way for FusionAuth to know that this request came from the server it expected (our express server).
+- The callback lambda is invoked after FusionAuth receives our request and responds. We’re expecting to receive an `access_token`, which will come through in that `body` argument. We’ll save the `access_token` to session storage and redirect back to the React client.
+
+Remember that request-followed-by-callback structure, because we’ll be using it again.
+
+We’re almost done! The only thing left is to set up the server-side session storage for our Access Token.  In order to do this, we need to configure `express-session`. The following settings work great for local testing, but you’ll probably want to check the `express-session` docs before moving to production.
+
+Add this to `server/index.js`
+
+```js
+// ...
+
+// configure Express app and install the JSON middleware for parsing JSON bodies
+const app = express();
+app.use(express.json());
+
+// configure sessions
+app.use(session(
+  {
+    secret: '1234567890',
+    resave: false,
+    saveUninitialized: false,
+    cookie: {
+      secure: 'auto',
+      httpOnly: true,
+      maxAge: 3600000
+    }
+  })
+);
+```
+
+
+
+##  Displaying user data
+Our React app looks for a user in `/user`. The Access Token that is granted to our Express server from FusionAuth isn’t human-readable, but we can pass it to FusionAuth’s `/introspect` endpoint to get a User Object (JSON like we showed earlier) from it. It's like saying 'Hey FusionAuth, you gave us this access token and so we can use that to access a user's data from you, because you trust your own access tokens.'
+
+We can get additional user-specific info from `/registration` as well, which is a FusionAuth specific API endpoint.  Then we can display whatever we want to the end user based on that user object (well, anything that the object gives us access to) which is what we are going to do now.
```

Comment:
```suggestion
We can get additional user-specific info from the `registration` endpoint as well, which is a FusionAuth specific API.  Then we can display whatever we want to the end user based on that user object (well, anything that the object gives us access to) which is what we are going to do now.
```

Reply:
These paths confuse me because they make me think that that is the path to the API call (which it is not).